### PR TITLE
WIP example config

### PR DESCRIPTION
**Pull request by beckermr**
Here is the start of an example config for this.

### CODE REVIEW

**Comment by beckermr** [comment]
@rmjarvis Care to review here?

File: configs/exp.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    ra:
+      type: Degrees
+      theta: { type: Random, min: "$ra_min_deg", max: "$ra_max_deg" }
+    dec:
+      type: Radians
+      theta:
+        type: RandomDistribution
+        function: "math.cos(x)"  # Uniform on the sky means P(delta) ~ cos(delta)
+        x_min: "$numpy.radians(dec_min_deg)"
+        x_max: "$numpy.radians(dec_max_deg)"
```
Comment (rmjarvis) [inline configs/exp.yaml:26]: This is allowed here, but it's probably better to put this in the `stamp` section.  Where it means the same thing, btw.  

After I made the `stamp` concept, I realized a few things that had been in `image` really belonged in `stamp`, including the positions, but I left them allowed in `image` for backwards compatibility.

File: configs/exp.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+psf:
+  type: RubinPSF
+
+gal:
+  type: Exponential
+  half_light_radius: 0.5
+  flux:
+    type: Eval
+    str: "10**(0.4*(30-mag))"
```
Comment (rmjarvis) [inline configs/exp.yaml:36]: Not sure where you're expecting to get mag from.  But right now, this won't work.  Probably should at least annotate a #TODO to say you need to figure out where the magnitudes come from.

File: configs/exp.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
+modules:
+  - seacliff
+  - galsim_extra
+
+image:
+  type: RubinCalexp
```
Comment (rmjarvis) [inline configs/exp.yaml:6]: Maybe mark the aspirational types as #TODO to remind us what's still left to be implemented.

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+galsim
+stackvana==0.*
```
Comment (rmjarvis) [inline requirements.txt:2]: Is this == restriction actually restrictive?  Are you indicating that once 1.0 is out, we don't want it?

File: seacliff/__init__.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
+try:
+    from importlib.metadata import version, PackageNotFoundError
+except ImportError:
+    from importlib_metadata import version, PackageNotFoundError
```
Comment (rmjarvis) [inline seacliff/__init__.py:4]: This is confusing.  Why the try block?

File: seacliff/rubin_sky_wcs.py [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    ----------
+    wcs : lsst.afw.geom.SkyWcs
+        The Rubin Sky WCS to be wrapped by this class.
+    origin : PositionD or PositionI or None, optional
+        If not None, the origin position of the image coordinate system. Note that
+        the conversion from 1-based to 0-based pixel indexing is **always** done.
+    """
+
+    # TODO: write a builder function for the config parser and settle on final API
+    _req_params = {"wcs": lsst.afw.geom.SkyWcs}
```
Comment (rmjarvis) [inline seacliff/rubin_sky_wcs.py:26]: TBH, if you're going to make an `lsst.aft.geom.SkyWcs` object be required, there's no point in having these `_req_params` (or `_opt_params`) at all.  They just allow for a quick description of how this could be read from a config dict without requiring any custom functions.  That mechanism doesn't know what a `SkyWcs` is, so this is useless.

Once we want this to be a config item, we'll have to write a custom loader that knows how to make a `SkyWcs`.

File: seacliff/tests/test_rubin_sky_wcs.py [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+    assert_allclose(ra1, ra2)
+    assert_allclose(dec1, dec2)
+
+    x1, y1 = wcs1.radecToxy(ra1, dec1, units=galsim.degrees)
+    x2, y2 = wcs2.radecToxy(ra2, dec2, units=galsim.degrees)
+    assert_allclose(x1, x2 - dx)
+    assert_allclose(y1, y2 - dy)
+
+    assert_allclose(x1, np.array([10, 9]))
+    assert_allclose(y1, np.array([78, 55]))
```
Comment (rmjarvis) [inline seacliff/tests/test_rubin_sky_wcs.py:67]: These test arrays are repeated a few times.  Probably worth making named variables at the top with these values.  (Especially in case some edge case comes along that we'd want to add to the arrays.)

File: seacliff/tests/test_rubin_sky_wcs.py [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+
+    _test_pos(wcs1, wcs6)
+
+
+def test_rubin_sky_wcs_pickles():
+    wcs1 = seacliff.RubinSkyWCS(_make_wcs(10), origin=galsim.PositionD(10, 2))
+    b = io.BytesIO()
+    pickle.dump(wcs1, b)
+    b.seek(0)
+    wcs2 = pickle.load(b)
```
Comment (rmjarvis) [inline seacliff/tests/test_rubin_sky_wcs.py:136]: It may be worth porting over our "do_pickle" test function that we have in GalSim, which tests ==, pickle, copy, deepcopy, and eval(repr()) all together.  We run this on all our objects.  Doing this kind of thing separately on every class gets old...
https://github.com/GalSim-developers/GalSim/blob/releases/2.3/tests/galsim_test_helpers.py#L399

**Review by rmjarvis — COMMENTED** [review]
Tests look good.  Most of my comments are on the config file, which I know is aspirational for now.  But it could probably use a big more commenting to describe the things that aren't totally obvious to the non-expert.

File: requirements.txt (hunk 4 above)
Comment (beckermr) [reply requirements.txt:2]: I've released dev versions of the stack (eg weeklies) as 0.2022.24 for the 2022_24 releases. Actual versions have full major versions. This pin installs the dev versions for ci testing and which is what we want.

File: seacliff/__init__.py (hunk 5 above)
Comment (beckermr) [reply seacliff/__init__.py:4]: This is for cross version compat for python below 3.8. 

See https://github.com/python/importlib_metadata

Comment (beckermr) [reply seacliff/__init__.py:4]: You may have missed the underscore versus dot. This package was only recently added to the stdlib.

Comment (rmjarvis) [reply seacliff/__init__.py:4]: Ah! Yes, I thought this was the same thing twice.  lol

**Comment by beckermr** [comment]
ok @rmjarvis - back to you!

**Review by rmjarvis — APPROVED** [review]
LGTM

**Comment by beckermr** [comment]
thank you!
